### DISCUSSION TITLE
Is there an equivalent to `execute-dir` in a single Quarto document (not a project)?

### DISCUSSION THREAD

**Discussion by turbanisch** [quarto-dev/quarto-cli#9667] [Q&A]
### Description

I would like to render a single Quarto document (not a Quarto Project) with the parent directory as working directory, ideally by simply clicking the "render" button in RStudio or invoking some other shortcut. Is there an equivalent to the `execute-dir` option of Quarto Projects for single Quarto documents?
I know I can specify the working directory with `quarto::quarto_render()` but this involves typing out both the document name and file path for each document.

**Comment by mcanouil** [comment]
`execute-dir` is a project level configuration, so you need to define `_quarto.yml` as documented (see <https://quarto.org/docs/projects/code-execution.html#working-dir>).

**Comment by cderv** [comment]
> Is there an equivalent to the execute-dir option of Quarto Projects for single Quarto documents?

There is no equivalent because `execute-dir` is only to change file root directory to project root directory. It does not allow to change to an arbitratry directory. 

So outside of project, it does not make sense. 

If you are using knitr engine document, you could leverage knitr options (independant of quarto) to change the execution directory. 

Though most of the time, it opens the door to side effect, and the solution is rather to handle path computation differently in chunks. 

Anyhow, changing execution directory for Quarto is a project level feature. Currently no way to set it for a single document, unless using `--execute-dir` command flag as you already suggested
